File: src/main/java/io/mfedirko/contactme/ContactHistory.java
package io.mfedirko.contactme;

import lombok.Builder;
import lombok.Getter;
import lombok.Value;
import lombok.extern.jackson.Jacksonized;

import java.time.LocalDateTime;

@Getter
@Builder
@Value
@Jacksonized
public class ContactHistory {
    LocalDateTime creationTimestamp;
    String fullName;
    String email;
    String messageBody;
}

File: src/main/kotlin/io/mfedirko/contactme/ContactHistory.kt
package io.mfedirko.contactme

import java.time.LocalDateTime
import kotlin.properties.Delegates

class ContactHistory {
    var creationTimestamp: LocalDateTime by Delegates.notNull()
    var id: Any by Delegates.notNull()
    var fullName: String by Delegates.notNull()
    var email: String by Delegates.notNull()
    var messageBody: String by Delegates.notNull()
    var status: Status by Delegates.notNull()

    enum class Status {
        /**
         * Initial status after submitting contact form
         */
        UNREAD,

        /**
         * Notification has been sent for this contact request
         */
        NOTIFIED
    }
}